integration/setup/setup_compat.go

```diff
@@ -163,15 +163,29 @@ func setupCompatCollections(tb testing.TB, ctx context.Context, client *mongo.Cl
 		_ = collection.Drop(ctx)
 
 		// if validators are set, create collection with them (otherwise collection will be created on first insert)
-		if validators := provider.Validators(*handlerF, collectionName); validators != nil {
+		if validators := provider.Validators(*handlerF, collectionName); len(validators) > 0 {
 			var opts options.CreateCollectionOptions
 			for key, value := range validators {
 				opts.SetValidator(bson.D{{key, value}})
 			}
 
-			// In this case, collection can't be created in MongoDB because MongoDB has a different validator format.
-			// So, we ignore errors.
-			database.CreateCollection(ctx, collectionName, &opts)
+			err := database.CreateCollection(ctx, collectionName, &opts)
+			switch err := err.(type) {
+			case nil:
+				// do nothing
+
+			case mongo.CommandError:
+				// If collection can't be created in MongoDB because MongoDB has a different validator format, it's ok.
+				// Otherwise, it's a problem:
+				if !strings.Contains(err.Message, `unknown top level operator: $tigrisSchemaString`) {
+					tb.Errorf("Failed to create collection %q: %v", collectionName, err)
+					tb.FailNow()
```

Comment:
```suggestion
					fallthrough
```

Reply:
@AlekSi that doesn't compile with "cannot fallthrough in type switch".

Reply:
`require.Contains` then?

Reply:
yeah, that looks better to me :)

Reply:
fixed